### PR TITLE
Implement structure for handling file permission

### PR DESCRIPTION
**Pull request by Einh06**
This PR sets the base structure to implement file move and deletion based on the current user permissions. It also refresh some of the comment on how the filestorage permissions are currently working.

### CODE REVIEW

File: schulcloud/FilesViewController.swift [hunk 1]
```diff
@@ -119,32 +118,48 @@ class FilesViewController: UITableViewController, NSFetchedResultsControllerDele
     }
 
     override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
-        guard let file = self.fetchedResultsController.sections?[indexPath.section].objects?[indexPath.row] as? File else { return false }
-        return file.permissions.contains(.write)
+        guard let currentUser = Globals.currentUser else { return false }
```
Comment (mathebox) [inline schulcloud/FilesViewController.swift:125]: It's currently possible to edit the three base folders and the course folders. Please add a guard statement for this.

**Review by mathebox — CHANGES_REQUESTED** [review]

File: schulcloud/FilesViewController.swift [hunk 2]
```diff
@@ -119,6 +119,10 @@ class FilesViewController: UITableViewController, NSFetchedResultsControllerDele
 
     override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
         guard let currentUser = Globals.currentUser else { return false }
+        guard let file = self.fetchedResultsController.sections?[indexPath.section].objects?[indexPath.row] as? File,
```
Comment (mathebox) [inline schulcloud/FilesViewController.swift]: please use `let file =  self.fetchedResultsController.object(at: indexPath)`

File: schulcloud/FilesViewController.swift [hunk 3]
```diff
@@ -119,6 +119,10 @@ class FilesViewController: UITableViewController, NSFetchedResultsControllerDele
...
+              file.id != FileHelper.rootDirectoryID, file.parentDirectory?.id != FileHelper.rootDirectoryID, file.parentDirectory?.id != FileHelper.userDirectoyID
```
Comment (mathebox) [inline schulcloud/FilesViewController.swift]: Are you sure about this? `file.parentDirectory?.id != FileHelper.userDirectoyID`
Doesn't this prevent the user from editing the files in this personal folder?

**Review by mathebox — CHANGES_REQUESTED** [review]
Please prevent editing course folders as well.

File: schulcloud/FilesViewController.swift (hunk 3 above)
Comment (Einh06) [reply schulcloud/FilesViewController.swift]: Omg, good catch, i meant to check the course.

**Pull request merged**